modules/kafka/kafka_test.go

```diff
@@ -1,26 +1,42 @@
 package kafka_test
 
 import (
+	"bytes"
 	"context"
+	"fmt"
+	"io"
 	"strings"
 	"testing"
 
 	"github.com/IBM/sarama"
+	"github.com/docker/docker/api/types/container"
 	"github.com/stretchr/testify/require"
 
 	"github.com/testcontainers/testcontainers-go"
 	"github.com/testcontainers/testcontainers-go/modules/kafka"
+	"github.com/testcontainers/testcontainers-go/network"
+	"github.com/testcontainers/testcontainers-go/wait"
 )
 
 func TestKafka(t *testing.T) {
-	topic := "some-topic"
+	const (
+		topic = "some-topic"
+		value = "kafka-message-value"
+	)
 
 	ctx := context.Background()
+	net, err := network.New(ctx)
+	require.NoError(t, err)
 
-	kafkaContainer, err := kafka.Run(ctx, "confluentinc/confluent-local:7.5.0", kafka.WithClusterID("kraftCluster"))
+	kafkaContainer, err := kafka.Run(ctx, "confluentinc/confluent-local:7.5.0", kafka.WithClusterID("kraftCluster"), network.WithNetwork(nil, net))
 	testcontainers.CleanupContainer(t, kafkaContainer)
 	require.NoError(t, err)
 
+	// Clean up the container after the test is complete
+	t.Cleanup(func() {
+		require.NoError(t, kafkaContainer.Terminate(ctx), "failed to terminate container: %v", err)
+	})
+
```

Comment:
bug: this isn't needed as that's what `CleanContainer` does.

Reply:
Excellent :champagne: